tests/test_integration.py

```diff
@@ -373,6 +379,15 @@ def test_scan_branch(test_positional_mor_deletes: Table) -> None:
     assert arrow_table["number"].to_pylist() == [1, 2, 3, 4, 6, 7, 8, 9, 10, 11, 12]
 
 
+@pytest.mark.integration
+def test_filter_on_new_column(test_table_add_column: Table) -> None:
+    arrow_table = test_table_add_column.scan(row_filter="b == '2'").to_arrow()
+    assert arrow_table["b"].to_pylist() == ['2']
```

Comment:
As a sanity check should we add a test which validates that a row_filter='b is NOT NULL' returns the same?

Reply:
That's a good one, added 👍